### PR TITLE
Feature - Add Button component

### LINKED ISSUE #139: Component - Button
# Description

Create the [Button component](https://www.figma.com/file/tXs6IdvpuX4jCLBWbJx0C6/Components-and-Patterns?node-id=6%3A15). Depending on the state, the component should follow the format and style guide shown in the wireframe.

- Create a Button component.
- The component will have two variations. A regular button component, and an anchor tag dressed as a button.
- The button props should include label. The label component should be a string, or a react component type, so that we can include icons like in the design. 
- The other props should include ones for styling. Use typescript to limit the different colours that can be used (purple, navy, gold, white).
- There are multiple variations in the designs that needs to be displayed in storybook, therefore the storybook actions should be used to do so.

# Assessment Criteria
  - [ ] Component has all states and follows the format and style guide.
  - [ ] Meets accessibility standards
  - [ ] Responsive (mobile, tablet, laptop)

### PR DESCRIPTION
**Pull request by mnigh**
Resolves #139.

### Notes
Because there are no designs for hover, active, focus, disabled and Hydrogen does not have yet have the ability to implement those states either (see [comment](https://github.com/GCTC-NTGC/gc-digital-talent/issues/139#issuecomment-879192715)), this PR does not include button states.

### CODE REVIEW

File: admin/resources/js/components/H2Components/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  onClick,
+}): React.ReactElement => {
+  return (
+    <button
+      type="button"
+      data-h2-border={`b(${borderColor}, all, solid, s)`}
+      data-h2-radius="b(s)"
+      data-h2-padding="b(top-bottom, xs) b(right-left, s)"
+      data-h2-bg-color={`b(${backgroundColor})`}
+      data-h2-font-family="b(sans)"
```
Comment (tristan-orourke) [inline admin/resources/js/components/H2Components/Button.tsx:39]: I'm unsure about including font-family on these H2 components. Shouldn't we just let them inherit, and set the font at a higher level?

File: admin/resources/js/components/H2Components/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    | "lightpurple"
+    | "[light]lightpurple[.1]"
+    | "darkpurple"
+    | "lightnavy"
+    | "[light]lightnavy[.1]"
+    | "darkgold"
+    | "[light]darkgold[.1]"
+    | "white"
+    | "[light]white[0]";
+  fontColor: "black" | "white" | "lightpurple" | "lightnavy" | "darkgold";
```
Comment (tristan-orourke) [inline admin/resources/js/components/H2Components/Button.tsx:21]: If we're being restrictive on colors, maybe we should just pass in a single color prop, and map that to different combinations.
```
color: "darkpurple" | "lightpurple" | "lightnavy" | "darknavy" | ...;

const colorMap = {
  darkpurple: {
    border: "darkpurple",
    background: "darkpurple",
    font: "white",
  },
  lightpurple: {
     border: "darkpurple",
    background: "lightpurple",
    font: "darkpurple",
  },
  ...
}

Comment (tristan-orourke) [reply admin/resources/js/components/H2Components/Button.tsx:21]: Or, now that I've looked closer at the stories, use two props:
```
color: "primary" | "secondary" | "cta" | "white;
style: "solid" | "outline" | "inline";

const colorMap = {
  purple_solid: {
    border: "darkpurple",
    background: "darkpurple",
    font: "white",
  },
 ...
}
```
Or, we could even export 12 different Button components, one for each style? 
Either way, if we only want certain combinations, lets set that up in this component.
(BTW, what's CTA?)

**Review by tristan-orourke — CHANGES_REQUESTED** [review]

File: admin/resources/js/components/H2Components/Button.tsx (hunk 1 above)
Comment (mnigh) [reply admin/resources/js/components/H2Components/Button.tsx:39]: It totally depends on if other fonts will be used on the project. If there's only ever one font used, then yes, inheritance from the html element is probably the easiest. Will that be the case?

**Review by tristan-orourke — APPROVED** [review]

**Pull request merged**